coins/ethereum/contracts/Router.sol

```diff
@@ -0,0 +1,101 @@
+//SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.8.0;
+
+import "./Schnorr.sol";
+
+contract Router is Schnorr {
+    // contract owner
+    address owner;
+
+    // nonce is incremented for each batch of transactions executed
+    uint256 nonce; 
+
+    // prevents re-entrancy
+    uint8 internal locked;
+
+    struct PublicKey {
+        uint8 parity;
+        bytes32 px;
+    }
+
+    // current aggregated validator public key 
+    PublicKey publicKey;
+
+    struct Transaction {
+        address to;
+        uint256 value;
```

Comment:
uint64 which is what Substrate will use would minimize memory yet require an additional multiplication to re-expand. Should still be cheaper.

Reply:
ether-rs didn't have a uint64 type last I checked :/ was forced to use uint256

Reply:
https://docs.rs/ethers-core/latest/ethers_core/abi/token/enum.Token.html

Reply:
... huh. I wonder if it's a technicality since all arguments are padded to the size of the words. Would a FixedArray work?

Reply:
`FixedArray` is for a vec of Tokens, could do `FixedBytes`, that'd probably work. shall I try it?